conf/nginx.conf

```diff
@@ -0,0 +1,23 @@
+server {
+    listen 80 default_server;
+    charset utf-8;
+    root /usr/share/nginx/html;
+    index index.html;
+
+    # Handle /_status endpoint with a 200 response once the container is up
+    location = /_status {
+        add_header Content-Type text/plain;
+        return 200 'Success!';
+    }
+
+    # When requesting static paths with known extensions, try them, and return a 404 if not found
+    location ~* .+\.(css|js|html|png|jpe?g|gif|bmp|ico|json|csv|otf|eot|svg|svgz|ttf|woff|woff2|ijmap|pdf|tif|map) {
+        try_files $uri $uri/ =404;
+    }
+
+    # When requesting a path without extension, try it, and return the index if not found
+    # This allows for client-side route handling
+    location / {
```

Comment:
What if there is no trailing slash, e.g. the request is for `https://whateverWeCallThisThing.hypothes.is`. Will this routing still work?

Blah, the age-old trailing-slash headache. :)

Reply:
Yeah, it works too, because even though you can type a URL which is only the domain with no path (or empty path), in practice that does not "exist". HTTP clients always send `/` as the path, which matches this config:

See this for example: this is this docker image being run on port 8765. The browser does not display any path, but you can see the actual request includes `/`.

![image](https://user-images.githubusercontent.com/2719332/234890525-4aceeed6-b323-46e2-8679-a19b014dc4d2.png)

Same happens if you try using `cURL`: The request is performed to `/` path, even if you don't explicitly include it:

```shell
{16:18}~ ➭ curl -v http://localhost:8765
*   Trying 127.0.0.1:8765...
* Connected to localhost (127.0.0.1) port 8765 (#0)
> GET / HTTP/1.1
> Host: localhost:8765
> User-Agent: curl/7.85.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Server: nginx/1.24.0
< Date: Thu, 27 Apr 2023 14:23:58 GMT
< Content-Type: text/html; charset=utf-8
< Content-Length: 362
< Last-Modified: Mon, 09 Jan 2023 13:40:07 GMT
< Connection: keep-alive
< ETag: "63bc1937-16a"
< Accept-Ranges: bytes
< 
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>UI component pattern library</title>
  <link rel="stylesheet" href="/styles/pattern-library.css">
</head>
<body>
  <div id="app"></div>
  <script type="module" src="/scripts/pattern-library.bundle.js"></script>
</body>
</html>
* Connection #0 to host localhost left intact
```

And explicitly including `/`:

```shell
{16:18}~ ➭ curl -v http://localhost:8765/
*   Trying 127.0.0.1:8765...
* Connected to localhost (127.0.0.1) port 8765 (#0)
> GET / HTTP/1.1
> Host: localhost:8765
> User-Agent: curl/7.85.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Server: nginx/1.24.0
< Date: Thu, 27 Apr 2023 14:24:21 GMT
< Content-Type: text/html; charset=utf-8
< Content-Length: 362
< Last-Modified: Mon, 09 Jan 2023 13:40:07 GMT
< Connection: keep-alive
< ETag: "63bc1937-16a"
< Accept-Ranges: bytes
< 
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>UI component pattern library</title>
  <link rel="stylesheet" href="/styles/pattern-library.css">
</head>
<body>
  <div id="app"></div>
  <script type="module" src="/scripts/pattern-library.bundle.js"></script>
</body>
</html>
* Connection #0 to host localhost left intact
```